x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -127,7 +137,17 @@ export function App({
       },
     });
 
+    const kbnUrlStateStorage = createKbnUrlStateStorage({
+      history,
+      useHash: core.uiSettings.get('state:storeInSessionStorage'),
+    });
+    const { stop: stopSyncingQueryServiceStateWithUrl } = syncQueryStateWithUrl(
+      data.query,
+      kbnUrlStateStorage
+    );
```

Comment:
I think we need to add some of these dependencies to the `useEffect` list explicitly- the linting rule is not currently being required, but should be.